### ISSUE TITLE
Deployment on Apple App Store

### ISSUE THREAD

**Issue by kunjee17** [Zaid-Ajaj/fsharp-weekly#4]
Hi, @Zaid-Ajaj I have a apple license available with me. And I have deployed application using Xamarin Studio. (Fabulous)

If you are OK with it. I ll deploy this application on AppStore. The only thing is it will go under my name as license is on my name.

**Comment by kunjee17** [comment]
Also, let me know where you need to deploy it. iPhone, MacOS, iPad etc. iPhone and iPad normally go together.

**Comment by Zaid-Ajaj** [comment]
Hello Kunjan! 

Yes please, feel free to publish it to the app store however you like :smile: on iPhone and iPad would be a good start 

> note that the icon and application label may need adjustment, because I messed them up in the play store

**Comment by kunjee17** [comment]
Built submitted. PR raised. :)

**Comment by kunjee17** [comment]
I have submitted application for review. Hopefully it will be App Store in couple of days. It will take at least 48 hours for sure. Once done. Will close this issue with App Store link.

**Comment by kunjee17** [comment]
@Zaid-Ajaj Build rejected with below message. I have no idea what to do in this case. :(

```
Guideline 4.2 - Design - Minimum Functionality


We found that the usefulness of your app is limited by the minimal amount of content or features it includes.

Next Steps

We encourage you to review your app concept and incorporate different content and features that are in compliance with the App Store Review Guidelines.

We understand that there are no hard and fast rules to define useful or entertaining, but Apple and Apple customers expect apps to provide a really great user experience. Apps should provide valuable utility or entertainment, draw people in by offering compelling capabilities or content, or enable people to do something they couldn't do before or in a way they couldn't do it before.
```

**Comment by SergejDK** [comment]
First part would be to try implement iOS specific feautres e.g. push notification, changing the theme.

Other possibility is that the content is from the web and normally the users could go to the website -  no need for the app (Apples words). If the data would not be directly from the website they should accept it.

Those are the two parts which could be possible for the rejection.
Or you could ask them directly what they mean in the context of this app.

**Comment by Zaid-Ajaj** [comment]
Damn! Apples thinks the application is too trivial to deserve a place in the App Store 😅 They certainly have a point, the target audience is quite limited (F# devs with an iPhone) but maybe this is what they are missing: the app is only not to be of use for *every* Apple user but rather just those F# devs who are interested in F# weekly. 

@SergejDK Not sure if adding push notifications would actually solve the problem because there isn't much added value: if you use the app once a week that is enough to keep up with the website.

Maybe as a feature, we could add more content:
 - tweets with #fsharp 
 - tweets with @FableCompiler
 - recently created F# repositories from github (tagged fsharp, fable etc.)

Not sure how the UI would look like though

**Comment by SergejDK** [comment]
You are right that the push notifications are not enough, then a simple pwa would be enough.

Most of the content you mentioned is already on the website. Would be great if there is an api which just gives back the data :) instead of webscraping. But we can build it in the app, too. For twitter and GitHub there are some libs we can use.

The UI part could be a "hamburger menu"(master detail page) or what i personally like more is a tabpage.
Possible names for the tabs could be: 
- General (Website without github & twitter)
- Twitter
- GitHub

In case of adding more menu points a master detail page would be better because it is more readable but if we say we won't have more than 4 tabs we can go with tabpage. 

(Maybe we should move this discussion into another Issue so this issue is only for the apple store.)

**Comment by kunjee17** [comment]
@Zaid-Ajaj @SergejDK I guess we should be focusing on features now. May be provide authentication, in form web view to view and bookmark page. Sharing with other bookmark technologies. I ll keep pushing application to app store as well with every new release in android. Hopefully we can have our day soon. 

@Zaid-Ajaj I guess we should start feature listing as issues as well. So, community (me, me, me...) can pitch in to complete them. Also, allow them to file feature request. 

A tweet and guideline on what to do would be great start.

**Comment by Zaid-Ajaj** [comment]
@SergejDK I really like the tabbed page idea, I was thinking of something similar! 

@kunjee17 Of course! we should open up some issues with exact features that we want to add.

Concretely this is what I am thinking right now:
 - The application has 4 tabs 
   - F# Weekly (F# icon): shows blog entries as they are shown now
       - clicking on a blog entry shows the extracted links with three options:
       - View (eye icon?): opens the links 
       - Share (share icon): opens up default sharing options to twitter, slack, etc.
       - Bookmark (star icon)
   - F# tweets (twitter icon): shows F# related tweets 
   - F# repos (github icon) shows recent/popular F# repos on github
   - Bookmarks (star icon): shows the blog entries that were bookmarked/stared

- application implements push notifications and notifies users when there is a new blog entry (still haven't looked into it thoroughly)

The only downside to this since I will start accepting PRs that change source code is that I will need to withdraw my submission to the Applied F# Challenge :(

**Comment by kunjee17** [comment]
@Zaid-Ajaj Ok. Don't accept PRs. Instead file issue and work on features. I ll keep updating my repo for Apple Store. I guess that way we are not breaking any rules.

**Comment by SergejDK** [comment]
@Zaid-Ajaj stay in the F# challenge. If you have questions feel free to ask. 

@kunjee17 maybe one day apple will accept ...

**Comment by Zaid-Ajaj** [comment]
@kunjee17 @SergejDK Thanks guys for the support :heart: I will see what I can do

**Comment by SergejDK** [comment]
@Zaid-Ajaj 
Hey Zaid, the F# challenge ended if I am not quite wrong. Are we allowed to make PR's to this repo?

**Comment by Zaid-Ajaj** [comment]
> Hey Zaid, the F# challenge ended if I am not quite wrong. Are we allowed to make PR's to this repo?

Sure thing, PRs are very much appreciated, it has been a while and I really wanted to update the app since trendy github repos is now broken
